### PR TITLE
Add Synchronized map for utils pkg

### PR DESCRIPTION
**Pull request by Mmduh-483**

### CODE REVIEW

File: pkg/synchronized-map/synchronized_map.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+)
+
+// SynchronizedMap a thread safe string to anything map
+type SynchronizedMap struct {
+	Items map[string]interface{}
+	sync.RWMutex // Read Write mutex, guards access to internal map.
+}
+
+// New creates a new synchronized map
+func New() *SynchronizedMap {
```
Comment (moshe010) [inline pkg/synchronized-map/synchronized_map.go]: change to NewSynchronizedMap

Comment (Mmduh-483) [reply pkg/synchronized-map/synchronized_map.go]: fixed

File: pkg/synchronized-map/synchronized_map.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package synchronized_map
```
Comment (moshe010) [inline pkg/synchronized-map/synchronized_map.go]: can you move the file to utils folder and be part of util pkg

Comment (Mmduh-483) [reply pkg/synchronized-map/synchronized_map.go]: moved

**Pull request merged**